consensus/core/src/base_committer.rs

```diff
@@ -427,29 +427,29 @@ mod base_committer_builder {
             }
         }
 
-        #[allow(unused)]
+        #[expect(unused)]
         pub(crate) fn with_wave_length(mut self, wave_length: u32) -> Self {
             self.wave_length = wave_length;
             self
         }
 
-        #[allow(unused)]
+        #[expect(unused)]
         pub(crate) fn with_leader_offset(mut self, leader_offset: u32) -> Self {
             self.leader_offset = leader_offset;
             self
         }
 
-        #[allow(unused)]
+        #[expect(unused)]
         pub(crate) fn with_round_offset(mut self, round_offset: u32) -> Self {
             self.round_offset = round_offset;
             self
         }
 
         pub(crate) fn build(self) -> BaseCommitter {
             let options = BaseCommitterOptions {
-                wave_length: DEFAULT_WAVE_LENGTH,
-                leader_offset: 0,
-                round_offset: 0,
+                wave_length: self.wave_length,
+                leader_offset: self.leader_offset,
+                round_offset: self.round_offset,
```

Comment:
This makes more sense if the other methods will not be unused at some point, but in the future please don't include such changes in PRs like this, which should only replace the `#[allow`